virtio-devices/src/block.rs

```diff
@@ -309,9 +309,13 @@ impl BlockEpollHandler {
                         read_avg = if read_avg == u64::MAX {
                             latency * LATENCY_SCALE
                         } else {
-                            read_avg
-                                + ((latency * LATENCY_SCALE) - read_avg)
-                                    / (read_ops_last + read_ops.0)
+                            // Cumulative average is guaranteed to be
+                            // positive if being calculated properly
+                            (read_avg as i64
+                                + ((latency * LATENCY_SCALE) as i64 - read_avg as i64)
+                                    / (read_ops_last + read_ops.0) as i64)
+                                .try_into()
+                                .unwrap()
                         };
```

Comment:
I realised about the signedness issue in my updated version in my PR:

```
                            read_avg.saturating_add_signed(
                                ((latency as i64)
                                    .saturating_mul(LATENCY_SCALE as i64)
                                    .saturating_sub(read_avg as i64))
                                    / (read_ops_last + read_ops.0) as i64
```

Do you think there is still scope for overflow?

Reply:
Not that I can think of. To overflow the calculation, you will need to have a latency  lager than `(u64::MAX-1)/LATENCY_SCALE` ms (that is roughly 58494 years). If we really worry about silent overflow, I can replace all `as i64` with `try_into().unwrap()`, so that any future overflows will be reported as an panic. I am fine either way.